lib/ember-test-helpers/test-module.js

```diff
@@ -269,6 +271,36 @@ export default Klass.extend({
     };
   },
 
+  /*
+    Builds a version of the passed in context that contains deprecation warnings
+    for accessing properties that exist on the module.
+  */
+  _buildDeprecatedContext: function(module, context) {
+    var deprecatedContext = Object.create(context);
+
+    var keysForDeprecation = Object.keys(module);
+
+    for (var i = 0, l = keysForDeprecation.length; i < l; i++) {
+      this._proxyDeprecation(module, deprecatedContext, keysForDeprecation[i]);
+    }
+
+    return deprecatedContext;
+  },
+
+  /*
+    Defines a key on an object to act as a proxy for deprecating the original.
+  */
+  _proxyDeprecation: function(obj, proxy, key) {
+    if (typeof proxy[key] === 'undefined') {
+      Object.defineProperty(proxy, key, {
+        get: function() {
+          Ember.deprecate('Accessing the test module property "' + key + '" from a callback is deprecated.', false, { id: 'ember-test-helpers.test-module.callback-context', until: '0.6.0' });
```

Comment:
I think the name spacing that we have done (and is done here) helps a bit. It at least says ember-test-helpers so you can know what to look for. Perhaps we should add a project option here, and add it to the API for the debug handlers feature so that various tools have a better chance at this...